docs/getting-started/architecture.mdx

```diff
@@ -5,59 +5,97 @@ sidebar_position: 2
+received from or forwarded to that specific connector.
+* **Connector Plugin** - interfaces with Conduit on one side, and with the standalone connector plugin on the
+other and facilitates the communication between them. A standalone connector plugin is a separate process that
+implements the interface defined in
+[conduit-connector-protocol](https://github.com/ConduitIO/conduit-connector-protocol) and provides the read/write
+functionality for a specific resource (e.g. a database).
+  
+#### Processor
+
+Processors are stateless components that operate on a single record and can execute arbitrary
+  actions before forwarding the record to the next node in the pipeline. A processor can also choose to drop a
+  record without forwarding it. They can be attached either to a connector or to a pipeline, based on that they are
+  either processing only records that flow from/to a connector or all records that flow through a pipeline.
+  * **Processor Plugin** - interfaces with Conduit on one side, and with the standalone processor plugin on the
+  other and facilitates the communication between them. A standalone processor plugin is a WASM binary that
+  implements the interface defined in
+  [conduit-processor-sdk](https://github.com/ConduitIO/conduit-processor-sdk) and provides the logic for processing
+  a record (e.g. transforming its content).
+
+### Pipeline Engine
+
+The pipeline engine consists of nodes that can be connected together with Go channels to form
+  a data pipeline.
+  
+#### Node
+
+A node is a lightweight component that runs in its own goroutine and runs as long as the incoming channel
+is open. As soon as the previous node stops forwarding records and closes its out channel, the current node also
+stops running and closes its out channel. This continues down the pipeline until all records are drained and the
+pipeline gracefully stops. In case a node experiences an error all other nodes will be notified and stop running
+as soon as possible without draining the pipeline.
+
+## Persistence layer
+
+This layer is used directly by the [Orchestration layer](#orchestration-layer) and indirectly by the [Core layer](#core-layer), and [Schema registry service](#schema-registry-service) (through stores) to persist data. It provides the functionality of creating transactions and storing, retrieving and deleting arbitrary data like configurations or state. 
+
+More info on [storage](/docs/features/storage).
+
+## Connector utility services
+
+### Schema registry service
+
+The schema service is responsible for managing the schema of the records that flow through the pipeline. It provides functionality to validate records against a schema and to infer a schema from a record. The schema is stored in the schema store and can be referenced by connectors and processors. This service runs separately from Conduit.
+
+More info on [Schema Registry](/docs/features/schema-registry).
```

Comment:
It might very well be personal preference, but I think it's better not to use informal words like "info" and use "information".